lib/utils/utf8.c

```diff
@@ -79,3 +84,34 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const char *str, size_t maxlen)
+{
+	size_t i = 0, nbyte = 0;
+	size_t len = strnlen(str, maxlen);
+
+	while (i < len) {
+		if (str[i] <= ASCII_CHAR && str[i] >= '\0') {
+			i++;
+			continue;
+		} else {
+			if (str[i] <= SEQUENCE_MAX_LEN_2_BYTE &&
+				str[i] >= SEQUENCE_MIN_LEN_2_BYTE) {
+				nbyte = 2;
+			} else if (str[i] <= SEQUENCE_MAX_LEN_3_BYTE &&
+					   str[i] >= SEQUENCE_MIN_LEN_3_BYTE) {
+				nbyte = 3;
+			} else if (str[i] <= SEQUENCE_MAX_LEN_4_BYTE &&
+					   str[i] >= SEQUENCE_MIN_LEN_4_BYTE) {
+				nbyte = 4;
+			} else {
+				return false;
+			}
```

Comment:
You should be able to do something like that to not only make it more readable, but also only apply 1 check per `if`, as the previous `if` will already have check the `&& ...` part
```suggestion
			if (str[i] <= SEQUENCE_MIN_LEN_2_BYTE) {
				/* ... */
			} else if (str[i] <= SEQUENCE_MAX_LEN_2_BYTE) {
				nbyte = 2;
			} else if (str[i] <= SEQUENCE_MAX_LEN_3_BYTE) {
				nbyte = 3;
			} else if (str[i] <= SEQUENCE_MAX_LEN_4_BYTE) {
				nbyte = 4;
			} else {
				return false;
			}
```

Reply:
I hope he can keep the most simple and readable. I may not be able to optimize it.

Reply:
Should "It". I think I typed it wrong.
(OMG)